src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
I am skeptical that double precision is an issue here. AFAIK, floating point numbers always represent a number with the maximum precision available, in this case 64-bit. Round-off errors can accumulate if you do multiple operations on numbers in sequence, eg multiplying floating numbers in a loop. But this is not the case here.

>I disagree that there's something wrong in principle with using integer math though; rounding errors can be avoided by multiplying before dividing, while overflow can be curbed by using a big enough type.
True, @piscisaureus , @bnoordhuis , you would have to resort to Multiply128 as there is no built-in 128-bit integer type.